src/libra_py/workflows/nbra/step3.py

```diff
@@ -46,6 +46,142 @@
+
+    Args:
+        data_dim (int): how many rows or columns in the vibronic Hamiltonian matrix. This will be an even number becuase the 
+                        number of alpha orbtials should equal the number of beta orbitals
+        cbm_(alpha/beta)_index (int): index of VBM (or HOMO) in the matrix of the vibronic Hamiltonian 
+                                      (row or column index). Note, this index is from 1 
+        (alpha/beta)_include (int): how many orbitals to include from the cbm_(alpha/beta)_index
+        excitation_type (int):  0: Make SDs with beta electrons excited
+                                1: Make SDs with alpha electrons excited
+                                2: Make two sets of SDs, one for beta and one for alpha electrons excited  
+    """
+
+    num_same_spin_orbitals = int( data_dim / 2 )
+
+    alpha_electrons = []
+    beta_electrons  = []
+
+    # Check for potential errors
+    if data_dim % 2 > 0:
+        print ("The dimensions of your vibronic Hamiltonian matrix (or whatever data you wish to pass to the function build_SD) must be even")
+        print ("Exiting now ..")
+        sys.exit(0)
+
+    if cbm_alpha_index > num_same_spin_orbitals:
+        print ("You must have the same number of alpha and beta spin-orbitals in your basis. The index of the CBM for the alpha spin-orbitals\
+ cannot be greater than 1/2 the total number of spin orbitals ")
+        print ("Exiting now")
+        sys.exit(0)
+
+    elif cbm_alpha_index <= 0:
+        print ("The index of the CBM for the alpha spin-orbitals must be > 0")
+        print ("Exiting now")
+        sys.exit(0)
+
+    elif cbm_beta_index <= num_same_spin_orbitals:
+        print ("You must have the same number of alpha and beta spin-orbitals in your basis. The index of the CBM for the beta spin-orbitals\
+ cannot be less than 1/2 the total number of spin orbitals ")
+        print ("Exiting now")
+        sys.exit(0)
+
+    elif cbm_beta_index > 2*num_same_spin_orbitals:
+        print ("The index of the CBM for the beta spin-orbitals must be < total number of spin-orbitals")
+        print ("Exiting now")
+        sys.exit(0)
+
+    if cbm_alpha_index + alpha_include + 2 > num_same_spin_orbitals:
+        print ("Cannot include more alpha spin-orbitals than there are")
+        print ("Including the maximum amount")
+        alpha_include = num_same_spin_orbitals - cbm_alpha_index - 1
+        print ("New value for alpha_include = ", alpha_include)
+
+    if cbm_beta_index + beta_include + 2 > 2*num_same_spin_orbitals:
+        print ("Cannot include more beta spin-orbitals than there are")
+        print ("Including the maximum amount")
+        beta_include = 2*num_same_spin_orbitals - cbm_beta_index - 1
+        print ("New value for beta_include = ", beta_include) 
+
+    # Make ground state SD
+    for i in range( cbm_alpha_index - alpha_include, cbm_alpha_index + 1):
```

Comment:
redundant style, please use x = list( range (a,b) ), here and in other similar places